Intermittent test failure: test_get_weekly_metric_data

Seen here https://app.circleci.com/pipelines/github/mozilla/experimenter/68200/workflows/c2ec29f0-2c12-4473-a052-32aee4405703/jobs/422575




```py
=========================== short test summary info ============================
FAILED experimenter/experiments/tests/test_models.py::TestNimbusExperiment::test_get_weekly_metric_data - AssertionError: {'retained': {'has_weekly_data': True, 'data': {'branch-[672 chars] {}}} != {'retained': {'data': {'branch-a': [({'lower': 140, 'upp[811 chars]lse}}
  {'retained': {'data': {'branch-a': [({'lower': 140,
                                        'significance': 'neutral',
                                        'upper': 160},
                                       None),
                                      ({'lower': 130,
                                        'significance': 'neutral',
                                        'upper': 150},
                                       None),
                                      ({'lower': 120,
                                        'significance': 'neutral',
                                        'upper': 140},
                                       None)],
                         'branch-b': [({'lower': 140,
                                        'significance': 'neutral',
                                        'upper': 160},
                                       {'avg_rel_change': 15,
                                        'lower': 10,
                                        'significance': 'neutral',
                                        'upper': 20}),
                                      ({'lower': 130,
                                        'significance': 'neutral',
                                        'upper': 150},
                                       {'avg_rel_change': 10,
                                        'lower': 5,
                                        'significance': 'neutral',
                                        'upper': 15}),
                                      ({'lower': 120,
                                        'significance': 'neutral',
                                        'upper': 140},
                                       {'avg_rel_change': 5,
                                        'lower': 0,
                                        'significance': 'neutral',
                                        'upper': 10})]},
                'has_weekly_data': True},
-  'search_count': {'data': {}, 'has_weekly_data': False}}
?                                                        ^

+  'search_count': {'data': {}, 'has_weekly_data': False},
?                                                        ^

+  'total_amazon_search_count': {'data': {}, 'has_weekly_data': False},
+  'urlbar_amazon_search_count': {'data': {}, 'has_weekly_data': False}}
========== 1 failed, 3029 passed, 27402 warnings in 382.66s (0:06:22) ==========
parallel: This job failed:
pytest --cov --cov-report term-missing
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-6348)

--- a/experimenter/experimenter/experiments/tests/test_models.py
+++ b/experimenter/experimenter/experiments/tests/test_models.py
@@ -1391,10 +1391,14 @@ class TestNimbusExperiment(TestCase):
     @mock_valid_outcomes
     def test_get_weekly_metric_data(self):
         application = NimbusExperiment.Application.DESKTOP
-        outcomes = Outcomes.by_application(application)
+        Outcomes.clear_cache()
+        desktop_outcome_1 = Outcomes.get_by_slug_and_application(
+            "desktop_outcome_1", application
+        )
         experiment = NimbusExperimentFactory.create(
-            application=application,
-            primary_outcomes=[outcomes[0].slug],
+            application=NimbusExperiment.Application.DESKTOP,
+            primary_outcomes=[desktop_outcome_1.slug],
+            secondary_outcomes=[],
         )
         branch_a = NimbusBranchFactory.create(
             experiment=experiment, name="Branch A", slug="branch-a"
@@ -2589,10 +2593,14 @@ class TestNimbusExperiment(TestCase):
     @mock_valid_outcomes
     def test_get_metric_data_returns_correct_data(self):
         application = NimbusExperiment.Application.DESKTOP
-        outcomes = Outcomes.by_application(application)
+        Outcomes.clear_cache()
+        desktop_outcome_1 = Outcomes.get_by_slug_and_application(
+            "desktop_outcome_1", application
+        )
         experiment = NimbusExperimentFactory.create(
             application=NimbusExperiment.Application.DESKTOP,
-            primary_outcomes=[outcomes[0].slug],
+            primary_outcomes=[desktop_outcome_1.slug],
+            secondary_outcomes=[],
         )
         branch_a = NimbusBranchFactory.create(
             experiment=experiment, name="Branch A", slug="branch-a"
@@ -2709,12 +2717,12 @@ class TestNimbusExperiment(TestCase):
         results_b = experiment.get_metric_data("enrollments", "all", "branch-b")
 
         data_a = (
-            results_b.get(outcomes[0].friendly_name, {})
+            results_b.get(desktop_outcome_1.friendly_name, {})
             .get("data", {})
             .get("overall", {})
         )
         data_b = (
-            results_a.get(outcomes[0].friendly_name, {})
+            results_a.get(desktop_outcome_1.friendly_name, {})
             .get("data", {})
             .get("overall", {})
         )

--- a/experimenter/experimenter/nimbus_ui/tests/test_views.py
+++ b/experimenter/experimenter/nimbus_ui/tests/test_views.py
@@ -3164,6 +3164,11 @@ class TestResultsEditBranchImagesView(AuthTestCase):
 
 @mock_valid_outcomes
 class TestResultsView(AuthTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        Outcomes.clear_cache()
+
     def test_render_to_response(self):
         experiment = NimbusExperimentFactory.create_with_lifecycle(
             NimbusExperimentFactory.Lifecycles.ENDING_APPROVE_APPROVE,
@@ -3280,10 +3285,13 @@ class TestResultsView(AuthTestCase):
 
     def test_results_view_relative_ui_properties(self):
         application = NimbusExperiment.Application.DESKTOP
-        outcomes = Outcomes.by_application(application)
+        desktop_outcome_1 = Outcomes.get_by_slug_and_application(
+            "desktop_outcome_1", application
+        )
         experiment = NimbusExperimentFactory.create(
             application=application,
-            primary_outcomes=[outcomes[0].slug],
+            primary_outcomes=[desktop_outcome_1.slug],
+            secondary_outcomes=[],
         )
         branch_a = NimbusBranchFactory.create(
             experiment=experiment, name="Branch A", slug="branch-a"

